doc/source/internals.rst

```diff
@@ -94,8 +94,74 @@ not check (or care) whether the levels themselves are sorted. Fortunately, the
 constructors ``from_tuples`` and ``from_arrays`` ensure that this is true, but
 if you compute the levels and labels yourself, please be careful.
 
+.. _ref-composition-pandas:
 
-.. _:
+Define Original Data Structures using pandas
+--------------------------------------------
+
+.. warning:: If you simply want to add some functionality to ``pandas``, the easiest
+   way is monkey-patching. See :ref:`Adding Features to your pandas Installation <ref-monkey-patching>`.
+
+This section describes how to define your original data structure which extends ``pandas`` functionality using `composition <http://en.wikipedia.org/wiki/Composition_over_inheritance>`_.
+
+Below example shows an original class which is mostly compatible with ``Series``.
+The class have a ``_series`` property to hold standard ``Series`` (composition), and defining ``__getattr__`` to delegate all the undefined properties / methods to it.
```

Comment:
The class has a ``_series`` property that holds a standard ``Series`` (composition), and defines ``__getattr__`` to delegate all undefined properties and methods to the series.